pkg/synthetictests/allowedalerts/allow_on_revision_change.go

```diff
@@ -0,0 +1,63 @@
+package allowedalerts
+
+import (
+	"context"
+	"time"
+
+	"github.com/openshift/origin/pkg/synthetictests/historicaldata"
+
+	operatorv1client "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	"github.com/openshift/origin/pkg/synthetictests/platformidentification"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type etcdRevisionChangeAllowance struct {
+	operatorClient   operatorv1client.OperatorV1Interface
+	startingRevision int
+}
+
+var allowedWhenEtcdRevisionChange = &etcdRevisionChangeAllowance{}
+
+func NewAllowedWhenEtcdRevisionChange(ctx context.Context, operatorClient operatorv1client.OperatorV1Interface) (*etcdRevisionChangeAllowance, error) {
+	etcd, err := operatorClient.Etcds().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	biggestRevision := 0
+	for _, nodeStatus := range etcd.Status.NodeStatuses {
+		if int(nodeStatus.CurrentRevision) > biggestRevision {
+			biggestRevision = int(nodeStatus.CurrentRevision)
+		}
+	}
+	return &etcdRevisionChangeAllowance{
+		operatorClient:   operatorClient,
+		startingRevision: biggestRevision,
+	}, nil
+}
+
+func (d *etcdRevisionChangeAllowance) FailAfter(alertName string, jobType platformidentification.JobType) time.Duration {
+	etcd, err := d.operatorClient.Etcds().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return historicaldata.DurationOrDie(6.022)
```

Comment:
not `getClosestPercentilesValues` ?

Reply:
> not getClosestPercentilesValues ?

When this fails, we want to know because it shouldn't fail.  and avagadros number is easy to search for and hopefully stands out to people.

Reply:
why not to change the method to return an error and consume it `failOrFlake` method? I think having a clear reason is better than some constant number.